### ISSUE TITLE
Does it work in mobile screen readers (TB/VO)?

### ISSUE THREAD

**Issue by vbarinov** [focus-trap/focus-trap#80]
I checked the demo with mobile TalkBack/VoiceOver and traps don't work in them.
Is it intended by design or just impossible to implement?

The only solution for me is to hide **all** elements on a page **except** the modal window by adding `aria-hidden="true"` to them.

Also, thanks for the great library!

**Comment by vadimhp** [comment]
maybe focus-trap should be adding aria-hidden to them which would probably make focus-trap the ultimate solution

**Comment by stefcameron** [comment]
Closing issues older than 2020. Please LMK if this is still a problem.

**Comment by stefcameron** [comment]
@Manviel Thanks for the update! The latest release is 10.1.1. Note that `^8.4.2` wouldn't permit that release. But most likely if it's an issue in 8.x, then it's still an issue in 10.x since no work was done specifically related to this.

Unfortunately, I have no way of testing this use case, however, and I can't say that this library really focuses on mobile. It's more directed to desktops and "hopefully" works on mobile. Mobile is a whole different ball of wax. I believe this library was originally invented to address the issue of having a Modal dialog sitting in the middle of a larger window, and wanting to "trap" the user in it so they couldn't click anywhere outside the Modal, since the idea of a Modal is to focus the user's attention on a smaller piece of UI. But while mobile has the concept of a Modal, it doesn't really have it in the same way -- unless maybe you're using a tablet with a larger screen and a mobile browser loading a desktop browser app...

In any case, when a mobile use case works, it's a nice bonus for sure. If you can install the latest version, and then dig into the code to see where things are falling apart, maybe there's something we can do to help make it better.

Note that having this library automatically add `aria-hidden=true` on "everything" outside the tap doesn't seem like something focus-trap should be doing (i.e. I'm not yet convinced that would be an appropriate focus-trap-based solution for the issue). Based on the reports here so far, it sounds like focus-trap simply doesn't trap focus at all in this use case (for reasons we should first determine) and adding these attributes would just "fake it" by temporarily removing all outside elements from the a11y tree. It feels more like something an a11y-focused modal component library should be responsible for doing, for example. focus-trap's mission is to trap focus.

**Comment by stefcameron** [comment]
@bakura10 I have no intention to re-open this issue and add any fixes to this library specifically targeted at supporting mobile devices. I simply do not have capacity, nor the will, to support mobile. Please see my comments on #1430 and let's continue the conversation there.
